db/sqlc/migrations/000002_sessions.up.sql

```diff
@@ -0,0 +1,128 @@
+-- The sessions table contains LNC session related information.
+CREATE TABLE IF NOT EXISTS sessions (
+    -- The auto incrementing primary key.
+    id INTEGER PRIMARY KEY,
+
+    -- The ID that was used to identify the session in the legacy KVDB store.
+    -- This is derived directly from the local_public_key. In order to avoid
+    -- breaking the API, we keep this field here so that we can still look up
+    -- sessions by this ID.
+    alias BLOB NOT NULL UNIQUE,
+
+    -- The session's given label.
+    label TEXT NOT NULL,
+
+    -- The session's current state.
+    state SMALLINT NOT NULL,
+
+    -- The session type.
+    type SMALLINT NOT NULL,
+
+    -- expiry is the time that the session will expire.
+    expiry TIMESTAMP NOT NULL,
+
+    -- The session's creation time.
+    created_at TIMESTAMP NOT NULL,
+
+    -- The time at which the session was revoked.
+    revoked_at TIMESTAMP,
+
+    -- The mailbox server address.
+    server_address TEXT NOT NULL,
+
+    -- Whether the connection to the server should not use TLS.
+    dev_server BOOLEAN NOT NULL,
+
+    -- The root key ID to use when baking a macaroon for this session.
+    macaroon_root_key BIGINT NOT NULL,
+
+    -- The passphrase entropy to use when deriving the mnemonic for this LNC
+    -- session.
+    pairing_secret BLOB NOT NULL,
+
+    -- The private key of the long term local static key for this LNC session.
+    local_private_key BLOB NOT NULL,
+
+    -- The public key of the long term local static key for this LNC session.
+    -- This is derivable from the local_private_key but is stored here since
+    -- the local public key was used to identify a session when the DB was KVDB
+    -- based and so to keep the API consistent, we store it here so that we can
+    -- still look up sessions by this public key.
+    local_public_key BLOB NOT NULL UNIQUE,
+
+    -- The public key of the long term remote static key for this LNC session.
+    remote_public_key BLOB,
+
+    -- Whether the privacy mapper should be used for this session.
+    privacy BOOLEAN NOT NULL,
+
+    -- An optional account ID that this session is linked to.
+    account_id BIGINT REFERENCES accounts(id) ON DELETE CASCADE,
+
+    -- The session ID of the first session in this linked session group. This
+    -- is nullable for the case where the first session in the group is being
+    -- inserted, and so we first need to insert the session before we know the
+    -- ID to use for the group ID.
+    group_id BIGINT REFERENCES sessions(id) ON DELETE CASCADE
+);
+
+CREATE INDEX IF NOT EXISTS sessions_type_idx ON sessions(type);
+
+-- The session_macaroon_permissions table contains the macaroon permissions
+-- that are associated with a session.
+CREATE TABLE IF NOT EXISTS session_macaroon_permissions (
```

Comment:
Hmm I think it makes sense to include an `id` field for this table, `session_feature_configs` & `session_privacy_flags` as it'll likely be useful in the future?

Reply:
ok so i think it makes sense to add the `id` for mac perms and caveats since those are lists and not necessarily unique but think for feature configs and priv flags, it does not make sense to add `id` since those will always have a unique <foreign-key:one-other-field> pair. 

sound good?